pkg/granted/request/close.go

```diff
@@ -0,0 +1,150 @@
+package request
+
+import (
+	"context"
+	"fmt"
+
+	"connectrpc.com/connect"
+	"github.com/common-fate/cli/printdiags"
+	"github.com/common-fate/clio"
+	"github.com/common-fate/grab"
+	"github.com/common-fate/granted/pkg/cfaws"
+	"github.com/common-fate/granted/pkg/cfcfg"
+	"github.com/common-fate/sdk/config"
+	"github.com/common-fate/sdk/eid"
+	accessv1alpha1 "github.com/common-fate/sdk/gen/commonfate/access/v1alpha1"
+	"github.com/common-fate/sdk/service/access/grants"
+	"github.com/common-fate/sdk/service/access/request"
+	identitysvc "github.com/common-fate/sdk/service/identity"
+	"github.com/urfave/cli/v2"
+)
+
+var closeCommand = cli.Command{
+	Name:  "close",
+	Usage: "Close an active Just-In-Time access to a particular entitlement",
+	Flags: []cli.Flag{
+		&cli.StringFlag{Name: "aws-profile", Required: false, Usage: "Close a JIT access for a particular AWS profile"},
+		&cli.StringFlag{Name: "request-id", Required: false, Usage: "Close a JIT access for a particular access request ID"},
+	},
+	Action: func(c *cli.Context) error {
+		profiles, err := cfaws.LoadProfiles()
+		if err != nil {
+			return err
+		}
+
+		accessRequestID := c.String("request-id")
+
+		if accessRequestID != "" {
+			ctx := c.Context
+
+			cfg, err := config.LoadDefault(ctx)
+			if err != nil {
+				return err
+			}
+
+			client := request.NewFromConfig(cfg)
+
+			getRes, err := client.GetAccessRequest(ctx, connect.NewRequest(&accessv1alpha1.GetAccessRequestRequest{
+				Id: accessRequestID,
+			}))
+			clio.Debugw("result", "getAccessRequest", getRes)
+			if err != nil {
+				return fmt.Errorf("failed to get access request: , %w", err)
+			}
+
+			// check if access request has grants that need to be closed
+			grants := getRes.Msg.AccessRequest.Grants
+
+			needsDeprovisioning := false
```

Comment:
I think it is safe to call CloseAccessRequest on an already closed request.
It may be best to leave this check for request still active out of the client side and just call close every time to avoid any possible bugs in future